### ISSUE TITLE
Feature request: toggle between "show changed files only" and "show all files" while reviewing PRs

### ISSUE THREAD

**Issue by NoriSte** [ovity/octotree#1067]
### Description
As a Pro user, when I'm reviewing PRs, I'd fancy toggle between the changed files only and the whole tree. This way, I can easily check that:
- compared to siblings, changed files respect naming patterns
- opening sibling files, changed files respect coding patterns patterns
- see how the content directory is evolving and decide to change nesting patterns
- find duplicates, or suggest moving code to another file/directory

Anyway: thanks, Octotree is an amazing product 😊

**Comment by danhgit** [comment]
Hi, thanks for the suggestion! Do you think a toggle button will do or will something more be needed? I looked at the use cases you mentioned and have the feeling this feature, if built, has to do more than just introducing a switch between all-files vs changed-files. For example, in the all-files mode, clicking a file will navigate to the code page, but that might not be what you want. But I could be wrong. Any clarification and further thought is appreciated.

**Comment by NoriSte** [comment]
Hi Danh! Sorry for the late, well...

## The dream
Opening a from the all-files mode should add a panel, similar to the ones showed by Github for every changed file in the "right" position:
- the new panel would not show any diff, only the content of the file
- the right position would be the position where GitHub would have added the panel if that file would have changed

This opens up to a lot of possible scenarios:
- what happens when I return to the changed-only view? Should the no-diff files close? IMO, no
- should I have the possibility of "closing" the no-diff files? IMO, yes
- should the opened no-diff files be persisted while reloading the page? Yes, why not
- should the files without changes be marked as "special" to be visually identified by the ones opened by GitHub? Why not
- should the opened files be collapsible? Yes
- could I add comments to the no-diff files? No, I'd use the panel to copy the file URL and use it to comment out the files with diff

Obviously, I'm thinking freely without caring about what's feasible and what not 😉

## The "good enough"
The toggle and the "open file" (opening the file in the GitHub's UI, not the raw file) button would be useful enough 😊

---

What do you think? 😊

**Comment by danhgit** [comment]
Thanks @NoriSte. This is a very interesting idea. Everything in the "dream" part makes sense to me. Couple of questions:

* Instead of showing a panel in the right place. What if it shows a popup with the code when you click the no-diff file? Would that be more or less (or much less) useful?
* Can you clarify this one: "the "open file" (opening the file in the GitHub's UI, not the raw file)". What is the open-file button? And does "GitHub's UI" mean the code page?

**Comment by NoriSte** [comment]
> Instead of showing a panel in the right place. What if it shows a popup with the code when you click the no-diff file? Would that be more or less (or much less) useful?

Yep with the option to "stick" it, it shouldn't disappear if I move the mouse out because the most important purpose is "comparing" it (not through a diff but through eyes) with the files changed in the PR

> What is the open-file button? And does "GitHub's UI" mean the code page?

You're right, sorry:
- the "the open-file button" is the name of the file in the list, My bad
- the "GitHub's UI" is the code page ([this](https://github.com/NoriSte/cypress-wait-until/blob/master/src/index.js) instead of the [raw file](https://raw.githubusercontent.com/NoriSte/cypress-wait-until/master/src/index.js))

**Comment by danhgit** [comment]
Thanks! We'll be working on this. If you think of anything else that could be improved about this, please do let me know.

**Comment by NoriSte** [comment]
Thank you ❤️ and have a nice weekend!!
